src/packages/core/collection/collection.context.ts

```diff
@@ -143,11 +154,48 @@ export class UmbCollectionContext<ItemType, FilterModelType extends UmbCollectio
 		}
 	}
 
-	// TODO: find better name
-	setFilter(filter: Partial<FilterModelType>) {
+	/**
+	 * Sets the filter for the collection and refreshes the collection.
+	 * @param {Partial<FilterModelType>} filter
+	 * @memberof UmbCollectionContext
+	 */
+	public setFilter(filter: Partial<FilterModelType>) {
 		this.#filter.next({ ...this.#filter.getValue(), ...filter });
 		this.requestCollection();
 	}
+
+	// Views
+	/**
+	 * Sets the current view.
+	 * @param {ManifestCollectionView} view
+	 * @memberof UmbCollectionContext
+	 */
+	public setCurrentView(view: ManifestCollectionView) {
+		this.#currentView.next(view);
+	}
+
+	/**
+	 * Returns the current view.
+	 * @return {ManifestCollectionView}
+	 * @memberof UmbCollectionContext
+	 */
+	public getCurrentView() {
+		return this.#currentView.getValue();
+	}
+
+	#setCurrentView() {
+		const currentUrl = new URL(window.location.href);
+		const lastPathSegment = currentUrl.pathname.split('/').pop();
+		const views = this.#views.getValue();
+		const viewMatch = views.find((view) => view.meta.pathName === lastPathSegment);
+
+		/* TODO: Find a way to figure out which layout it starts with and set _currentLayout to that instead of [0]. eg. '/table'
+			For document, media and members this will come as part of a data type configuration, but in other cases "users" we should find another way. 
+			This should only happen if the current layout is not set in the URL.
```

Comment:
Maybe we could hardcode the one we look for? Just in the case for Users... so that looks for the grid view? Base on something that we find relatively stable.. then we might later make that configurable or smarter in some way?

Reply:
I have added a task for this, so we can come back to it.